packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/InvokeStaticMethodStatement.cs

```diff
@@ -7,13 +7,30 @@
 
 namespace Microsoft.Generator.CSharp.Statements
 {
-    public sealed record InvokeStaticMethodStatement(CSharpType? MethodType, string MethodName, IReadOnlyList<ValueExpression> Arguments, IReadOnlyList<CSharpType>? TypeArguments = null, bool CallAsExtension = false, bool CallAsAsync = false) : MethodBodyStatement
+    public sealed class InvokeStaticMethodStatement : MethodBodyStatement
     {
+        public CSharpType? MethodType { get; }
+        public string MethodName { get; }
+        public IReadOnlyList<ValueExpression> Arguments { get; }
+        public IReadOnlyList<CSharpType>? TypeArguments { get; }
```

Comment:
maybe this does not have to be nullable because empty array could mean exactly that this method invocation does not have type arguments.